tcms/testcases/migrations/0009_populate_missing_text_history.py

```diff
@@ -19,6 +19,24 @@ def forward_copy_data(apps, schema_editor):
             history.save()
 
 
+def reverse_copy_data(apps, schema_editor):
+    test_case_model = apps.get_model('testcases', 'TestCase')
+    historical_test_case_model = apps.get_model('testcases', 'HistoricalTestCase')
+
+    for test_case in test_case_model.objects.all():
+        history = historical_test_case_model.objects.filter(
+            case_id=test_case.pk).order_by('-history_id').first()
+
+        # In 0006_merge_text_field_into_testcase_model we may have
+        # failed to save the text into the history record leaving
+        # historical records with text == None.
+        # If the TC was not modified since then we try to fix the last
+        # historical record
+        if history.text == test_case.text:
+            history.text = historical_test_case_model().text
+            history.save()
```

Comment:
This is totally out of context and is being handled in #1824. Remove the last commit.

Reply:
Also it is incorrect.